core/controlplane/config/templates/cloud-config-controller

```diff
@@ -2711,14 +2743,31 @@ write_files:
     content: {{.AssetsConfig.EtcdTrustedCA}}
 {{ end }}
 
+  - path: /etc/kubernetes/proxy-kubeconfig.yaml
+    content: |
+        apiVersion: v1
+        kind: Config
+        clusters:
+        - name: local
+          cluster:
+            server: http://127.0.0.1:8080
+        users:
+        - name: kubelet
```

Comment:
I think it should use service acccount token to have same permissions granted by RBAC

Reply:
Can we access the API server on the controller nodes via `localhost` _and_ use the service account cert at the same time? I don't think the certificate includes `localhost` (or `127.0.0.1`). (I'll give it a try nevertheless)

Reply:
you can volunteer to auth requests when talking to insecure APIServer port. That is what kube-controller-manager is doing: runs using serviceaccount, but uses insecure port to communicate with apiserver